appservice/src/deploy/runDeployTask.ts

```diff
@@ -27,7 +27,8 @@ export async function tryRunPreDeployTask(context: IActionContext, deployFsPath:
     let preDeployTaskResult: IPreDeployTaskResult = { taskName, exitCode: undefined, failedToFindTask: false };
 
     if (taskName && shouldExecuteTask(scmType, settingKey, taskName)) {
-        const task: vscode.Task | undefined = await findTask(deployFsPath, taskName);
+        const tasks: vscode.Task[] = await vscode.tasks.fetchTasks();
+        const task: vscode.Task | undefined = taskUtils.findTask(deployFsPath, taskName, tasks);
```

Comment:
Should you just include `fetchTasks()` as part of `findTask`?  I always seem them called together anyway.